packages/core/src/components/ModalNew/ModalTopActions/ModalTopActions.types.ts

```diff
@@ -0,0 +1,17 @@
+import React from "react";
+import MenuButton from "../../MenuButton/MenuButton";
+import IconButton from "../../IconButton/IconButton";
+import { ButtonColor } from "../../Button/ButtonConstants";
+
+export type ModalTopActionsColor = "light" | "dark";
+export type ModalTopActionsButtonColor =
+  | ButtonColor.PRIMARY
+  | ButtonColor.ON_PRIMARY_COLOR
+  | ButtonColor.ON_INVERTED_BACKGROUND;
+
+export interface ModalTopActionsProps {
+  renderAction?: (color?: ModalTopActionsButtonColor) => React.ReactElement<typeof MenuButton | typeof IconButton>;
```

Comment:
@shlomitc suggests for it to be
```suggestion
  renderAction?: React.ReactElement<typeof MenuButton | typeof IconButton>;
```

let's discuss it.
The main motivation for making it a function is the following:
```typescript
// color on `renderHeaderAction` is "light"
<Modal closeButtonTheme="light" renderHeaderAction={(color) => <IconButton ...props color={color} />
```

vs inlining it and let the user put the color twice
```typescript
<Modal closeButtonTheme="light" renderHeaderAction={<IconButton ...props color={"light"} />
```

Reply:
I think we can support both, if you want the color you can use a function and use it in your component, otherwise you can control it yourself with just passing a component